lib/shard/src/segment_holder/mod.rs

```diff
@@ -947,7 +947,7 @@ impl SegmentHolder {
             // Get segment to snapshot
             let segment = match proxy_segment {
                 LockedSegment::Proxy(proxy_segment) => {
-                    proxy_segment.read().wrapped_segment.clone().get()
+                    proxy_segment.read().wrapped_segment.get_cloned()
```

Comment:
Effectively the only place (outside tests) where we actually need the owned `Arc` by cloning it.